packages/calcite-components/src/utils/floating-ui.ts

```diff
@@ -53,6 +53,138 @@ async function patchFloatingUiForNonChromiumBrowsers(): Promise<void> {
+ *
+ * **Note:** exported for testing purposes only
+ *
+ * @param root0
+ * @param root0.referenceEl
+ * @param root0.floatingEl
+ * @param root0.overlayPositioning
+ * @param root0.placement
+ * @param root0.flipDisabled
+ * @param root0.flipPlacements
+ * @param root0.offsetDistance
+ * @param root0.offsetSkidding
+ * @param root0.arrowEl
+ * @param root0.type
+ * @param component
+ * @param root0.referenceEl.referenceEl
+ * @param root0.referenceEl.floatingEl
+ * @param root0.referenceEl.overlayPositioning
+ * @param root0.referenceEl.placement
+ * @param root0.referenceEl.flipDisabled
+ * @param root0.referenceEl.flipPlacements
+ * @param root0.referenceEl.offsetDistance
+ * @param root0.referenceEl.offsetSkidding
+ * @param root0.referenceEl.arrowEl
+ * @param root0.referenceEl.type
+ * @param component.referenceEl
+ * @param component.floatingEl
+ * @param component.overlayPositioning
+ * @param component.placement
+ * @param component.flipDisabled
+ * @param component.flipPlacements
+ * @param component.offsetDistance
+ * @param component.offsetSkidding
+ * @param component.arrowEl
+ * @param component.type
+ */
+export const positionFloatingUI =
+  /* we export arrow function to allow us to spy on it during testing */
+  async (
+    component: FloatingUIComponent,
+    {
+      referenceEl,
+      floatingEl,
+      overlayPositioning = "absolute",
+      placement,
+      flipDisabled,
+      flipPlacements,
+      offsetDistance,
+      offsetSkidding,
+      arrowEl,
+      type
+    }: {
+      referenceEl: ReferenceElement;
+      floatingEl: HTMLElement;
+      overlayPositioning: Strategy;
+      placement: LogicalPlacement;
+      flipDisabled?: boolean;
+      flipPlacements?: EffectivePlacement[];
+      offsetDistance?: number;
+      offsetSkidding?: number;
+      arrowEl?: SVGElement;
+      type: UIType;
+    }
+  ): Promise<void> => {
+    if (!referenceEl || !floatingEl) {
+      return null;
+    }
+
+    await floatingUIBrowserCheck;
+
+    const {
+      x,
+      y,
+      placement: effectivePlacement,
+      strategy: position,
+      middlewareData
+    } = await computePosition(referenceEl, floatingEl, {
+      strategy: overlayPositioning,
+      placement:
+        placement === "auto" || placement === "auto-start" || placement === "auto-end"
+          ? undefined
+          : getEffectivePlacement(floatingEl, placement),
+      middleware: getMiddleware({
+        placement,
+        flipDisabled,
+        flipPlacements,
+        offsetDistance,
+        offsetSkidding,
+        arrowEl,
+        type
+      })
+    });
+
+    if (arrowEl && middlewareData.arrow) {
+      const { x, y } = middlewareData.arrow;
+      const side = effectivePlacement.split("-")[0] as Side;
+      const alignment = x != null ? "left" : "top";
+      const transform = ARROW_CSS_TRANSFORM[side];
+      const reset = { left: "", top: "", bottom: "", right: "" };
+
+      if ("floatingLayout" in component) {
```

Comment:
should we use `Object.hasOwn(component, "floatingLayout)` here instead of `in` so we don't need spend extra time checking component's prototype chain? Or do we care about inherited properties in this scenario?

Reply:
Great observation. I think we could look at our `in` usage across the board. Maybe we could use ESLint to help enforce this (haven't found an existing rule at the moment 🤔). 

Currently, I'm not too worried about the prototype lookup, but it'd be good to make this change for the future.

WRT `hasOwn`, I think we could to wait for Safari 17 before we can start using it. Unless we use `hasOwnProperty` and then switch to `hasOwn` along with https://eslint.org/docs/latest/rules/prefer-object-has-own.